promotion/drain.go

```diff
@@ -374,6 +386,81 @@ func (service *Service) RedeemAndTransferFunds(ctx context.Context, credentials
 		*wallet.UserDepositAccountProvider)
 }
 
+func redeemAndTransferGeminiFunds(
+	ctx context.Context,
+	service *Service,
+	wallet *walletutils.Info,
+	total decimal.Decimal,
+) (*walletutils.TransactionInfo, error) {
+
+	ns := uuid.Must(uuid.FromString(wallet.UserDepositDestination))
```

Comment:
Is there a reason we want to have code that could panic? Since this functions returns an error, you can return the error from `uuid.FromString()` if it fails.